node/router/src/heartbeat.rs

```diff
@@ -117,32 +154,16 @@ pub trait Heartbeat<N: Network>: Outbound<N> {
             return;
         }
 
-        // Retrieve the trusted peers.
-        let trusted = self.router().trusted_peers();
-        // Retrieve the bootstrap peers.
-        let bootstrap = self.router().bootstrap_peers();
-
-        // Find the oldest connected peer, that is neither trusted nor a bootstrap peer.
-        let oldest_peer = self
-            .router()
-            .get_connected_peers()
-            .iter()
-            .filter(|peer| !trusted.contains(&peer.ip()) && !bootstrap.contains(&peer.ip()))
-            .filter(|peer| !self.router().cache.contains_inbound_block_request(&peer.ip())) // Skip if the peer is syncing.
-            .filter(|peer| self.is_block_synced() || self.router().cache.num_outbound_block_requests(&peer.ip()) == 0) // Skip if you are syncing from this peer.
-            .min_by_key(|peer| peer.last_seen())
-            .map(|peer| peer.ip());
-
-        // Disconnect from the oldest connected peer, if one exists.
-        if let Some(oldest) = oldest_peer {
+        // Disconnect from the oldest connected peer, which is the first entry in the list
+        // of removable peers.
+        // Do nothing, if the list is empty.
+        if let Some(oldest) = self.get_removable_peers().into_iter().map(|peer| peer.ip()).next() {
```

Comment:
Really appreciate the addition of the comment here! I don't mean to nit but this might be a little more idiomatic:
```suggestion
        if let Some(oldest) = self.get_removable_peers().first().map(Peer::ip) {
```

I might also suggest we add a unit test for `get_removable_peers` as getting the order wrong there would lead to a more unstable network? I'm not sure how involved test writing might be here, we could perhaps include this in future improvements to our test coverage, wdyt?

Reply:
Well now you're just showing off! 😛

And yes since `get_removable_peers` peers is not influenced by timeouts, it can be unit tested.

Reply:
I'm surprised there isn't a lint this specific error. 

Let me look into how involved unit tests would be, but I generally agree that they are a good idea here.